### PR TITLE
Foxy controller stopper

### PR DESCRIPTION
**Pull request by fmauch**
Backport of #309 

This is currently a draft, as I haven't tested it, yet.

### CODE REVIEW

**Comment by fmauch** [comment]
@urmahp would you like helping me out in testing this? This currently compiles fine, but running it crashes.

**Comment by urmahp** [comment]
> @urmahp would you like helping me out in testing this? This currently compiles fine, but running it crashes on this line:
> 
> https://github.com/UniversalRobots/Universal_Robots_ROS2_Driver/blob/b2014e060493509b6a9a848dfd30d0c55670da42/ur_controllers/src/gpio_controller.cpp#L203
> 
> Seems like `state_interfaces_[StateInterfaces::PROGRAM_RUNNING]` is not returning a valid object. I am not entirely sure why this happens, though.

I found the issue, the intialized flag is not added to the state interfaces in the hardware interface on the foxy branch, however it is on the main branch, see [here](https://github.com/UniversalRobots/Universal_Robots_ROS2_Driver/blob/main/ur_robot_driver/src/hardware_interface.cpp#L213-L214). This means that the when we try to index program_running the index is out of range and therefore the program crashes. 

I see two possible fixes adding the initialized flag and using it during controller startup (which i believe is where it is used) or simply removing it which will also solve the problem. However I prefer the first one. I can fix it if you agree?

**Comment by fmauch** [comment]
> I see two possible fixes adding the initialized flag and using it during controller startup (which i believe is where it is used) or simply removing it which will also solve the problem. However I prefer the first one. I can fix it if you agree?

I agree, the first one would be preferable. Thank you @urmahp

**Comment by fmauch** [comment]
Aweseome, thanks @urmahp I'll update this PR to be reviewable and then we can proceed with this.

File: ur_controllers/src/gpio_controller.cpp [hunk 1]
```diff
@@ -100,6 +100,10 @@ controller_interface::InterfaceConfiguration ur_controllers::GPIOController::sta
   for (size_t i = 0; i < 11; ++i) {
     config.names.emplace_back("gpio/safety_status_bit_" + std::to_string(i));
   }
+  config.names.emplace_back("system_interface/initialized");
```
Comment (fmauch) [inline ur_controllers/src/gpio_controller.cpp:103]: That would probably require adding this to the xacro, as well. (For Foxy we decided to stick with the one in this repo)

Comment (fmauch) [reply ur_controllers/src/gpio_controller.cpp:103]: resolved in [7ea0d27](https://github.com/UniversalRobots/Universal_Robots_ROS2_Driver/pull/324/commits/7ea0d27263aa4bcba4f059a3db4582e1221ffe71)

**Pull request merged**